### PR TITLE
Change bearby balance function

### PR DESCRIPTION
**Pull request by Ben-Rey**

### CODE REVIEW

**Review by peterjah — APPROVED** [review]

File: src/bearbyWallet/BearbyAccount.ts [hunk 1]
```diff
@@ -69,30 +68,22 @@ export class BearbyAccount implements IAccount {
...
-    // get node url: This is a temporary solution. We should get the balance from the provider
-    const nodeUrl = await getNodesUrl();
 
-    const addressInfos = await postRequest<BalanceResponse>(nodeUrl, body);
+    const res = await web3.massa.getAddresses(this._address);
 
-    if (addressInfos.isError || addressInfos.error) {
-      throw addressInfos.error.message;
+    if (!res && !res.result) {
+      throw new Error(res.error.message);
```
Comment (peterjah) [inline src/bearbyWallet/BearbyAccount.ts:75]: did you test this?? are we sure res.error.message is always defined?

File: src/bearbyWallet/BearbyAccount.ts [hunk 2]
```diff
@@ -69,30 +68,22 @@ export class BearbyAccount implements IAccount {
...
     }
 
+    // TODO: check if we need to cast to AddressInfo. Fix to do on bearby.js
+    const addressInfo = res?.result[0] as AddressInfo;
```
Comment (peterjah) [inline src/bearbyWallet/BearbyAccount.ts:79]: res.result[0]

Comment (Ben-Rey) [reply src/bearbyWallet/BearbyAccount.ts:79]: if i remove as AddressInfo, final_balance and candidate_balance are not reconized

Comment (Ben-Rey) [reply src/bearbyWallet/BearbyAccount.ts]: If you were only talking about the "?", fixed it 🙏

**Review by thomas-senechal — APPROVED** [review]

**Pull request merged**